RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -257,20 +270,51 @@ public Optional<ObjectNode> getSnapshotStatus(
         }
     }
 
-    public Mono<BulkResponse> sendBulkRequest(String indexName, String body, IRfsContexts.IRequestContext context) {
+    Retry getBulkRetryStrategy() {
+        return Retry.backoff(10, Duration.ofSeconds(2)).maxBackoff(Duration.ofSeconds(60));
+    }
+
+    public Mono<BulkResponse> sendBulkRequest(String indexName, List<BulkDocSection> docs, IRfsContexts.IRequestContext context) {
         String targetPath = indexName + "/_bulk";
+        Map<String, BulkDocSection> docsMap = docs.stream().collect(Collectors.toMap(d -> d.getDocId(), d -> d));
```

Comment:
Can we defer constructing this docsMap to be inside the mono.defer? Possibly through an AtomicReference?